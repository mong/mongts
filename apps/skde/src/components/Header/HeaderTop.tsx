import { NavigateNextRounded, ArrowBack } from "@mui/icons-material";
import {
  Breadcrumbs,
  Breakpoint,
  Link,
  Toolbar,
  Typography,
  styled,
  Container,
} from "@mui/material";
import Grid from "@mui/material/Grid2";
import { useTheme } from "@mui/material/styles";
import { useMediaQuery } from "@mui/material";

const StyledToolbar = styled(Toolbar)(({ theme }) => ({
  backgroundColor: theme.palette.background.default,
  paddingTop: theme.spacing(5),
  paddingBottom: theme.spacing(5),
}));

const LogoImage = styled("img")(({ theme }) => ({
  [theme.breakpoints.down("lg")]: {
    height: 40,
    width: 100,
  },
  [theme.breakpoints.up("lg")]: {
    height: 50,
    width: 125,
  },
}));

const StyledBreadcrumbSeparator = styled(NavigateNextRounded)(({ theme }) => ({
  color: theme.palette.primary.light,
}));

type BreadCrumbStop = {
  link: string;
  text: string;
};

export type BreadCrumbPath = {
  path: BreadCrumbStop[];
};

const SkdeBreadcrumbs = (props: BreadCrumbPath) => {
  const { path } = props;

  const theme = useTheme();
  const onMobile = useMediaQuery(theme.breakpoints.down("md"));

  if (onMobile && path.length > 1) {
    const secondToLastElement = path.at(-2);
    return (
      <Breadcrumbs
        separator={<ArrowBack fontSize="medium" />}
        aria-label="breadcrumb"
      >
        <div></div>
        <Link
          underline="hover"
          key="mobile_breadcrumb"
          href={secondToLastElement.link}
          variant="h6"
        >
          {secondToLastElement.text}
        </Link>
      </Breadcrumbs>
    );
  }
  return (
    <Breadcrumbs
      separator={<StyledBreadcrumbSeparator fontSize="large" />}
      aria-label="breadcrumb"
    >
      {path.slice(0, -1).map((row, index) => (
        <Link underline="hover" key={index} color="inherit" href={row.link}>
          {row.text}
        </Link>
      ))}
      <Typography color="text.primary">{path.at(-1).text}</Typography>
    </Breadcrumbs>
  );
};

type HeaderTopProps = {
  breadcrumbs: BreadCrumbPath;
  maxWidth?: false | Breakpoint;
};

export const HeaderTop = (props: HeaderTopProps) => {
  const { breadcrumbs, maxWidth } = props;

  return (
    <StyledToolbar className="header-top">
      <Container maxWidth={maxWidth ? maxWidth : false} disableGutters={true}>
        <Grid container spacing={2}>
          <Grid size={{ xs: 12 }}>
            <Link href={"https://www.skde.no/"}>
              <LogoImage src="/img/logos/logo-skde.svg" alt="SKDE logo" />
            </Link>
          </Grid>
          <Grid size={{ xs: 12 }}>
            <SkdeBreadcrumbs path={breadcrumbs.path} />
          </Grid>
        </Grid>
      </Container>
    </StyledToolbar>
  );
};
