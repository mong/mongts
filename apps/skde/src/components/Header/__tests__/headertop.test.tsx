import React from "react";
import { render } from "@testing-library/react";
import { HeaderTop } from "../HeaderTop";
import { BreadCrumbPath } from "../HeaderTop";
import { describe, it, expect } from "vitest";

describe("HeaderTop component", () => {
  it("renders with breadcrumbs", () => {
    const breadcrumbs: BreadCrumbPath = {
      path: [
        { text: "Home", link: "/" },
        { text: "About", link: "/about" },
      ],
    };
    const { container } = render(<HeaderTop breadcrumbs={breadcrumbs} />);
    expect(container).toMatchSnapshot();
  });

  it("renders with maxWidth prop", () => {
    const maxWidth = "lg";
    const { container } = render(
      <HeaderTop
        maxWidth={maxWidth}
        breadcrumbs={{ path: [{ text: "Home", link: "/" }] }}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
