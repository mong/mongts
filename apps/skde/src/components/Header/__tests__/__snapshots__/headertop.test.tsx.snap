// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HeaderTop component > renders with breadcrumbs 1`] = `
<div>
  <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular header-top css-rtbc6q-MuiToolbar-root"
  >
    <div
      class="MuiContainer-root MuiContainer-disableGutters css-1aa3h9d-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-2 css-zjhuaj-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-109nfct-MuiGrid-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1orsvah-MuiTypography-root-MuiLink-root"
            href="https://www.skde.no/"
          >
            <img
              alt="SKDE logo"
              class="css-tkwvp9"
              src="/img/logos/logo-skde.svg"
            />
          </a>
        </div>
        <div
          class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
        >
          <nav
            aria-label="breadcrumb"
            class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-5cvx94-MuiTypography-root-MuiBreadcrumbs-root"
          >
            <ol
              class="MuiBreadcrumbs-ol css-1uwp4ue-MuiBreadcrumbs-ol"
            >
              <li
                class="MuiBreadcrumbs-li"
              >
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-nv1n36-MuiTypography-root-MuiLink-root"
                  href="/"
                >
                  Home
                </a>
              </li>
              <li
                aria-hidden="true"
                class="MuiBreadcrumbs-separator css-jhr02i-MuiBreadcrumbs-separator"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1i7gzfm-MuiSvgIcon-root"
                  data-testid="NavigateNextRoundedIcon"
                  focusable="false"
                  font-weight="bold"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9.31 6.71c-.39.39-.39 1.02 0 1.41L13.19 12l-3.88 3.88c-.39.39-.39 1.02 0 1.41s1.02.39 1.41 0l4.59-4.59c.39-.39.39-1.02 0-1.41L10.72 6.7c-.38-.38-1.02-.38-1.41.01"
                  />
                </svg>
              </li>
              <li
                class="MuiBreadcrumbs-li"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-4leci6-MuiTypography-root"
                >
                  About
                </p>
              </li>
            </ol>
          </nav>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`HeaderTop component > renders with maxWidth prop 1`] = `
<div>
  <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular header-top css-rtbc6q-MuiToolbar-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-ho17r4-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-2 css-zjhuaj-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-109nfct-MuiGrid-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1orsvah-MuiTypography-root-MuiLink-root"
            href="https://www.skde.no/"
          >
            <img
              alt="SKDE logo"
              class="css-tkwvp9"
              src="/img/logos/logo-skde.svg"
            />
          </a>
        </div>
        <div
          class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
        >
          <nav
            aria-label="breadcrumb"
            class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-5cvx94-MuiTypography-root-MuiBreadcrumbs-root"
          >
            <ol
              class="MuiBreadcrumbs-ol css-1uwp4ue-MuiBreadcrumbs-ol"
            >
              <li
                class="MuiBreadcrumbs-li"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-4leci6-MuiTypography-root"
                >
                  Home
                </p>
              </li>
            </ol>
          </nav>
        </div>
      </div>
    </div>
  </div>
</div>
`;
